Sources/MessagingPushAPN/rich push/RichPushRequestHandler.swift

```diff
@@ -0,0 +1,40 @@
+import CioMessagingPush
+import CioTracking
+import Foundation
+#if canImport(UserNotifications)
+import UserNotifications
+
+internal class RichPushRequestHandler {
+    static let shared = RichPushRequestHandler()
+
+    @Atomic private var requests: [String: RichPushRequest] = [:]
+
+    private init() {}
+
+    func startRequest(
+        _ request: UNNotificationRequest,
+        content: PushContent,
+        completionHandler: @escaping (UNNotificationContent) -> Void
+    ) {
+        let requestId = request.identifier
+
+        let existingRequest = requests[requestId]
+        if existingRequest != nil { return }
+
+        let newRequest = RichPushRequest(pushContent: content, request: request, completionHandler: completionHandler)
+        requests[requestId] = newRequest
+
+        newRequest.start()
+    }
+
+    func stopAll() {
+        requests.forEach { item in
+            let request = item.value
```

Comment:
I created the `let` out of habit. Since `item` is a tuple key/value pair I usually name the `item.value` to a variable name that makes more sense in the context. In this scenario, value == request. 

But for this very simple loop, no need.